tabular/src/autogluon/tabular/task/tabular_prediction/predictor_v2.py

```diff
@@ -0,0 +1,733 @@
+from autogluon.core.utils.savers import save_pkl
+from autogluon.core.utils.utils import setup_outputdir, setup_compute, setup_trial_limits, default_holdout_frac
+
+from .dataset import TabularDataset
+from .hyperparameter_configs import get_hyperparameter_config
+from .predictor import TabularPredictor
+from .presets_configs import set_presets, unpack
+from ...features import AutoMLPipelineFeatureGenerator
+from ...learner import AbstractLearner, DefaultLearner
+from ...metrics import get_metric
+from ...trainer import AbstractTrainer, AutoTrainer
+
+
+logger = logging.getLogger()  # return root logger
+
+
+class TabularPredictorV2(TabularPredictor):
+    """
+    AutoGluon Task for predicting values in column of tabular dataset (classification or regression)
+    """
+    Dataset = TabularDataset
+    predictor_file_name = 'predictor.pkl'
+
+    def __init__(
+            self,
+            label,
+            problem_type=None,
+            output_directory=None,
+            eval_metric=None,
+            stopping_metric=None,
+            feature_metadata=None,
+            verbosity=2,
+            id_columns=None,
+            label_count_threshold=10,
+            cache_data=True,
+            feature_generator=None,
+            learner_type=DefaultLearner,
+            trainer_type=AutoTrainer,
+            random_seed=0,
+
+            # learner_kwargs=None -> could simplify
+
+        ):
+        self._learner = None
+        self._trainer = None
+        self.output_directory = setup_outputdir(output_directory)
+        self.problem_type = problem_type
+        self.eval_metric = eval_metric
+        self.label_column = label
+        self.feature_metadata = None
+        self.class_labels = None  # TODO: method
+        self.class_labels_internal = None  # TODO: method
+        self.class_labels_internal_map = None  # TODO: method
+        self.stopping_metric = stopping_metric
+        self.feature_metadata = feature_metadata  # TODO: Unused, FIXME: currently overwritten after .fit, split into two variables: one for pre and one for post processing.
+        self.verbosity = verbosity  # TODO: Unused
+        self.id_columns = id_columns
+        if self.id_columns is None:
+            self.id_columns = []
+        self.label_count_threshold = label_count_threshold  # TODO: Requires num_bagging_folds to be set to properly work, move num_bagging_folds into init?
+        self.cache_data = cache_data
+        self.feature_generator = feature_generator  # TODO: Fix presets
+        self.learner_type = learner_type
+        self.trainer_type = trainer_type
+        self.random_seed = random_seed
+        self.has_learner = False
+
+    @unpack(set_presets)
+    def fit(self,
+            train_data,
+            tuning_data=None,
+            time_limits=None,
+            presets=None,
+            eval_metric=None,
+            stopping_metric=None,
+            auto_stack=False,
+            hyperparameter_tune=False,
```

Comment:
I found `num_trials` is sufficient to switch HPO on/off, `hyperparameter_tune == Ture` is equivalent to `num_trials > 1`

Reply:
How does time_limits play into this? If someone want to train for 10 minutes, then they might not care about setting num_trials, and num_trials is less obviously linked to hyperparameter_tune. I foresee most users specifying time_limits rather than num_trials, and few if any specifying both.